### PR TITLE
Add Blazor support for table row IDs and validity

### PR DESCRIPTION
**Pull request by mollykreis**
# Pull Request

## 🤨 Rationale

Add Blazor support for changes made in #953, which includes:
- `id-field-name` attribute on the table
- `validity` property on the table, which is exposed through an async method because properties have to be retrieved through JS interop
- `checkValidity()` function on the table, which is exposed through an async method because it uses JS interop

## 👩‍💻 Implementation

- Add items listed above to the Blazor component
- Create an `ITableValidity` interface to expose the readonly table validity state.
- Create internal `TableValidity` class to provide the functionality needed to convert the javascript validity object to a C# object

## 🧪 Testing

- Write new auto test for `id-field-name`
- Manually tested that `CheckValidityAsync()` and `GetValidityAsync()` return the correct values

## ✅ Checklist

- [ ] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

**Review by msmithNI — APPROVED** [review]

File: packages/nimble-blazor/NimbleBlazor/wwwroot/NimbleBlazor.lib.module.js
Comment (rajsite) [inline packages/nimble-blazor/NimbleBlazor/wwwroot/NimbleBlazor.lib.module.js:72]: Are we sure this is working? I would imagine this needs to be JSON.stringify before passing?

We need some playwright tests lol. Should we add something to the sample apps to run this? maybe like a button to press that runs it) as a manual test?

Comment (mollykreis) [reply packages/nimble-blazor/NimbleBlazor/wwwroot/NimbleBlazor.lib.module.js:72]: I did see this working locally by putting some code in the sample app. I debated leaving that code in as part of this PR, but after some discussion with @atmgrifter00, we decided not to. But, we both completely agreed with the idea that we need playwright tests sooner rather than later.

I can go ahead and put my test code back into the sample app unless there are any objections.

**Review by atmgrifter00 — APPROVED** [review]

File: packages/nimble-blazor/NimbleBlazor/wwwroot/NimbleBlazor.lib.module.js
Comment (rajsite) [reply packages/nimble-blazor/NimbleBlazor/wwwroot/NimbleBlazor.lib.module.js:72]: I don't feel strongly on way or another. If we do think we can start working on playwright tests and we capture this as a test that needs to be written that sounds fine. Long-term I don't think it's a particularly interesting in the example app, just (abusing?) it for a spot to put a manual test. If tested locally and y'all already agreed that we don't need it there sounds good to me 👍

**Review by rajsite — APPROVED** [review]

**Comment by mollykreis** [comment]
Note: There will be some minor changes to this code based on #980. I'm planning on waiting until that PR is merged and the updating this PR with the changes to the `TableValidity` field names.

File: packages/nimble-blazor/NimbleBlazor/wwwroot/NimbleBlazor.lib.module.js
Comment (mollykreis) [reply packages/nimble-blazor/NimbleBlazor/wwwroot/NimbleBlazor.lib.module.js:72]: I'm planning to leave it as-is for now (nothing in the example to check validity), but I did manually double check that those functions are working now that I've made the rename from _rowId_ to _recordId_.

**Pull request merged**
